server/src/main/java/org/elasticsearch/index/mapper/SourceValueFetcher.java

```diff
@@ -58,7 +58,13 @@ public SourceValueFetcher(String fieldName, SearchExecutionContext context, Obje
             while (queue.isEmpty() == false) {
                 Object value = queue.poll();
                 if (value instanceof List) {
-                    queue.addAll((List<?>) value);
+                    for (Object o : (List<?>) value) {
+                        if (o != null) {
+                            queue.add(o);
+                        } else if (nullValue != null) {
```

Comment:
Could we add a test to cover this case where the mappings provide a `null_value`? I'm not totally sure this logic is needed (it may be handled already in `SourceLookup#extractValue`).

Reply:
There is one in FieldFetcherTests?  The part that checks `nullable_long_field`.

Reply:
Sorry about that, I read before coffee :) I am able to remove this 'else' block around `nullValue` and still the test passes. Do we need to strengthen the test, or can this be removed?

Reply:
Oh, interesting, it looks as though `extractValue` already handles this case so we can remove it.